transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,229 @@
+	MainScriptRelPath string
+	RequirementsTxt   string
+	IsDjangoProject   bool
+}
+
+// Init Initializes the transformer
+func (t *PythonDockerfileGenerator) Init(tc transformertypes.Transformer, env *environment.Environment) (err error) {
+	t.Config = tc
+	t.Env = env
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *PythonDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// findMainScripts returns the path of .py files having the main function
+func findMainScripts(pythonFilesPath []string) ([]string, error) {
+	if len(pythonFilesPath) == 0 {
+		return []string{}, nil
+	}
+	var pythonMainFiles []string
+	for _, pythonFilePath := range pythonFilesPath {
+		pythonFile, err := ioutil.ReadFile(pythonFilePath)
+		if err != nil {
+			logrus.Warnf("Error in reading python file %s: %s.", pythonFilePath, err)
+		} else {
+			if strings.Contains(string(pythonFile), pythonMain1) || strings.Contains(string(pythonFile), pythonMain2) {
+				pythonMainFiles = append(pythonMainFiles, pythonFilePath)
+			}
+		}
+	}
+	if len(pythonMainFiles) == 0 {
+		return []string{}, fmt.Errorf("could not find the main function in python files")
+	}
+	return pythonMainFiles, nil
+}
+
+// findDjangoDependency checks for django dependency in the requirements.txt file
+func findDjangoDependency(reqTxtFilePath string) bool {
+	reqTxtFile, err := ioutil.ReadFile(reqTxtFilePath)
+	if err != nil {
+		logrus.Warnf("Error in reading the file %s: %s.", reqTxtFilePath, err)
+		return false
+	}
+	return strings.Contains(string(reqTxtFile), django)
+}
+
+// getMainPythonFileForService returns the main file used by a service
+func getMainPythonFileForService(mainPythonFilesPath []string, baseDir string, serviceName string) string {
+	var mainPythonFilesRelPath []string
+	for _, mainPythonFilePath := range mainPythonFilesPath {
+		if mainPythonFileRelPath, err := filepath.Rel(baseDir, mainPythonFilePath); err == nil {
+			mainPythonFilesRelPath = append(mainPythonFilesRelPath, mainPythonFileRelPath)
+		}
+	}
+	return qaengine.FetchSelectAnswer(common.ConfigServicesKey+common.Delim+serviceName+common.Delim+common.ConfigMainPythonFileForServiceKeySegment, fmt.Sprintf("Select the main file to be used for the service %s :", serviceName), []string{fmt.Sprintf("Selected main file will be used for the service %s", serviceName)}, mainPythonFilesRelPath[0], mainPythonFilesRelPath)
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PythonDockerfileGenerator) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
+	pythonFiles, err := common.GetFilesByExtInCurrDir(dir, []string{pythonExt})
+	if err != nil {
+		logrus.Errorf("Error while finding python files %s", err)
+		return nil, nil
+	}
+	if len(pythonFiles) == 0 {
+		return nil, nil
+	}
+	mainPythonFilesPath, err := findMainScripts(pythonFiles)
+	if err != nil {
+		return nil, err
```

Comment:
Will python programs always need to have a main part? If there is no main should we ask users which is the starting python file in transform function from all the py files in the base project directory?